### DISCUSSION TITLE
"colour.CAM16_to_XYZ" divides by zero when C = 0.

### DISCUSSION THREAD

**Discussion by KrisKennaway** [colour-science/colour#904] [Q&A]
With 0.3.16 and python 3.9.7:

>>> import colour.appearance.cam16 as cam16
>>> import numpy as np
>>> specification = cam16.CAM_Specification_CAM16(J=41.731207905126638, C=0.0, h=217.067959767393010)
>>> XYZ_w = np.array([95.05, 100.00, 108.88])
>>> L_A = 318.31
>>> Y_b = 20.0
>>> cam16.CAM16_to_XYZ(specification, XYZ_w, L_A, Y_b)
/Volumes/Stuff/apple2/dither/venv/lib/python3.9/site-packages/colour/appearance/ciecam02.py:1504: RuntimeWarning: divide by zero encountered in double_scalars
  P_1 = ((50000 / 13) * N_c * N_cb * e_t) / t
array([ 19.01427392,  19.99891973,  21.76421405])

i.e. temporary_magnitude_quantity_inverse returns 0 when C = 0

I am not sure if this is just a cosmetic issue (i.e. not correctness) but it would be nice to suppress the warning if it's intentional.

**Comment by thomasmansencal** [comment]
Hi @KrisKennaway,

So this is intentional, we try as much as possible to not hide anything like that so that the user can decide on the best course of action.

It is possible to granularly filter all the warnings using the [colour.utilities.filter_warnings](https://colour.readthedocs.io/en/develop/generated/colour.utilities.filter_warnings.html#colour.utilities.filter_warnings) definition or the related [colour.utilities.suppress_warnings](https://colour.readthedocs.io/en/develop/generated/colour.utilities.filter_warnings.html#colour.utilities.suppress_warnings) context manager.

**Comment by KrisKennaway** [comment]
Thanks for the quick response.  I guess my confusion is that C=0 is a
perfectly valid input right?  Technically working with t = inf might give
the right result in that case, although I would be concerned about
potential numerical instabilities from small nonzero values where t becomes
arbitrarily large.

I wonder if it would be better to work with 1/t instead of t here?  Then I
think all values should be finite.
